src/site/scripts/components/profile/actionbutton.tsx

```diff
@@ -116,77 +118,40 @@ export class ActionButton extends React.Component<IActionButtonProps, IActionBut
      */
     private renderConfirmation(): JSX.Element {
         let onClick: () => void;
-        let value: string;
 
-        if (!this.state.expanded || this.state.delay > 0) {
+        if (!this.state.expanded) {
             onClick = undefined;
-            value = this.state.delay ? this.state.delay.toString() : "";
         } else {
             onClick = (): void => {
                 this.props.action();
                 this.toggleExpansion();
             };
-            value = this.state.confirmation;
         }
 
         return (
             <div className="action-confirmation">
-                <input
-                    onClick={onClick}
-                    type="button"
-                    value={value} />
-                    {this.renderCancel()}
+                <div className="action-confirmation-overlay"/>
+                <ConfirmationDialog
+                    action={() => onClick()}
+                    onCancel={() => this.toggleExpansion()}
```

Comment:
Nit: can you manually annotate `: void` in all these lambdas? I like this as a rule because it helps avoid accidentally triggering unexpected behavior when returning the value of functions to things that react to return values.